js/slideshow.js

```diff
@@ -180,6 +181,7 @@
 		 * @param {Object} next
 		 */
 		next: function (current, next) {
+			this.container.find('.icon-loading-dark').show();
```

Comment:
You could move this before [line 99](https://github.com/nextcloud/gallery/pull/165/commits/b0efcfad58bb9d5c2dbc9b2f0ff14c492fceb943#diff-da2846a114690f2d9e41a05c414598a7R99).

I think it then looks more natural to call `show()` and then call `hide()` on resolution of the promise.